### PR TITLE
S4 add products page

### PR DESCRIPTION
**Pull request by APaltrow**
Task: S4 - Products page - products list
- Added UI for Products Page 
- Configured JSON with products
- Added products to mock api
- Made Call to API and displayed products

### CODE REVIEW

File: src/components/Products/ProductsItem/ProductsItem.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                <li>Brand</li>
+                <li>Delivery</li>
+                <li>Stock</li>
+              </ul>
+              <ul className={style.additional_info}>
+                <li>{originCountry}</li>
+                <li>{brands.join(', ')}</li>
+                <li>{delivery.area.join(', ')}</li>
+                <li>{`${stock.amount} ${stock.measure}`}</li>
+              </ul>
```
Comment (Yuliia143) [inline src/components/Products/ProductsItem/ProductsItem.tsx]: It isn't a good idea to have 2 lists
You always need to be careful with the order of values
+ Think it's better to create an object and go through it and render this list as one

File: src/components/Products/ProductsItem/ProductsItem.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            </div>
+          </Accordion>
+        </div>
+
+        <div className={style.price_and_delivery_container}>
+          <div className={style.price}>
+            <p>{`${productPrice} ${price.currency}`}</p>
+
+            {price.discount ? <span>{price.amount.toFixed(2)}</span> : null}
```
Comment (Yuliia143) [inline src/components/Products/ProductsItem/ProductsItem.tsx]: Logic in JSX + magic number + think you will use the same on the product page

File: src/components/ui/InfoTooltip/InfoTooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { FC } from 'react';
+
+import style from './InfoTooltip.module.scss';
+
+interface InfoTooltipProps {
+  info: string | number;
```
Comment (Yuliia143) [inline src/components/ui/InfoTooltip/InfoTooltip.tsx]: Is it a component for amount of products ?
Then why string or number ?

File: src/types/productTypes.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  area: string[];
+}
+
+export enum BuyBy {
+  PSC = 'pcs',
+  KGS = 'kgs',
+  BOX = 'box',
+  PACK = 'pack',
+}
+
```
Comment (Yuliia143) [inline src/types/productTypes.ts]: Think it's better to have file for each type in this case

File: src/components/Products/ProductsItem/ProductsItem.tsx (hunk 1 above)
Comment (APaltrow) [reply src/components/Products/ProductsItem/ProductsItem.tsx]: Joined 2 lists into 1

File: src/components/Products/ProductsItem/ProductsItem.tsx (hunk 2 above)
Comment (APaltrow) [reply src/components/Products/ProductsItem/ProductsItem.tsx]: Removed logic from JSX and set magic number to a constant

File: src/components/ui/InfoTooltip/InfoTooltip.tsx (hunk 3 above)
Comment (APaltrow) [reply src/components/ui/InfoTooltip/InfoTooltip.tsx]: It is a ui component which will be reused on single product page. it contains non number values such as ' - 30%' etc.
However, number type does not make sense here so I left string only and updated relevant components.

File: src/types/productTypes.ts (hunk 4 above)
Comment (APaltrow) [reply src/types/productTypes.ts]: Decomposed the types to separate files.

File: src/constants/price.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export const PRICE_DESCIMALS = 2;
```
Comment (Yuliia143) [inline src/constants/price.ts:1]: decimals*

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**